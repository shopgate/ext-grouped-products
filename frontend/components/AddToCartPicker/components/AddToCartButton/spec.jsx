import React from 'react';
import { mount } from 'enzyme';
import configureStore from 'redux-mock-store';
import PickerAddToCartButton from './index';

jest.mock('@shopgate/pwa-ui-shared/AddToCartButton', () =>
  require.requireActual('./AddToCartButton.mock'));

const openListSpy = jest.fn();
const mockConditioner = { check: jest.fn(() => new Promise(resolve => resolve(true))) };

const buttonProps = {
  openList: openListSpy,
  isDisabled: false,
  isLoading: false,
  addedQuantity: 0,
  noShadow: true,
  conditioner: mockConditioner,
};

describe('<PickerAddToCartButton />', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });
  it('should render as expected', () => {
    const component = mount(<PickerAddToCartButton {...buttonProps} />);
    expect(component).toMatchSnapshot();
  });
  // it('should execute the clickhandler as expected', () => {
  //   const component = mount(<PickerAddToCartButton {...buttonProps} />);
  //   component.find('button').simulate('click');
  //   expect(openListSpy).toHaveBeenCalledTimes(1);
  // });
});
