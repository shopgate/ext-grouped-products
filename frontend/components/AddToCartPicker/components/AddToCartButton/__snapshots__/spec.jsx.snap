// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PickerAddToCartButton /> should render as expected 1`] = `
<PickerAddToCartButton
  addedQuantity={0}
  conditioner={
    Object {
      "check": [MockFunction],
    }
  }
  isDisabled={false}
  isLoading={false}
  noShadow={true}
  openList={[MockFunction]}
>
  <button
    data-test-id="addToCartButton"
    onClick={[Function]}
  />
</PickerAddToCartButton>
`;
